intg-appletv/tv.py

```diff
@@ -680,33 +680,66 @@ async def _poll_worker(self) -> None:
         while self._atv is not None:
             update = {}
 
-            # Push updates are not reliable for power events, and if the device is in standby it reports state idle!
-            if self._is_feature_available(FeatureName.PowerState):
-                # Off isn't sent with push updates with the current pyatv library
-                # Care must be taken to not override certain states like playing and paused
-                if self._atv.power.power_state == PowerState.Off:
-                    # The Off state is important to wakeup the device in the command handler
-                    update["state"] = self._atv.power.power_state
-                elif self._atv.power.power_state == PowerState.On and self._state not in (
-                    DeviceState.Playing,
-                    DeviceState.Paused,
-                    DeviceState.Stopped,
-                    DeviceState.Seeking,
-                    DeviceState.Loading,
-                ):
-                    update["state"] = self._atv.power.power_state
+            await self._handle_power(update)
 
             if self._is_feature_available(FeatureName.App) and self._atv.metadata.app.name:
                 update["source"] = self._atv.metadata.app.name
-                if playing := await self._atv.metadata.playing():
-                    update["position"] = playing.position if playing.position else 0
-                    update["total_time"] = playing.total_time if playing.total_time else 0
+
+            if data := await self._atv.metadata.playing():
+                update["position"] = data.position if data.position else ""
+                update["total_time"] = data.total_time if data.total_time else ""
+                update["title"] = data.title if data.title else ""
+                update["artist"] = data.artist if data.artist else ""
+                update["album"] = data.album if data.album else ""
+                if data.media_type is not None:
+                    update["media_type"] = data.media_type
+
+                match data.repeat:
+                    case RepeatState.Off:
+                        update["repeat"] = "OFF"
+                    case RepeatState.All:
+                        update["repeat"] = "ALL"
+                    case RepeatState.Track:
+                        update["repeat"] = "ONE"
+
+                if data.shuffle is not None:
+                    update["shuffle"] = data.shuffle != ShuffleState.Off
+
+                # off state is not included in metadata, don't override it
+                if update["state"] != PowerState.Off:
+                    self._state = data.device_state
+                    update["state"] = data.device_state
+
+                await self._process_artwork(update)
```

Comment:
The artwork processing is now called every 10 seconds while something is playing.
Fortunately the `self._atv.metadata.artwork` function uses an internal cache and usually doesn't involve a network request.
However, it still involves base64 encoding and the data is also sent in every `entity_change` event. This could trigger image reloading in the UI app and also cause excessive logging on the device. Base64 data is already filtered out in the logs of the integration, the Python API wrapper and core service, but I'll check the UI.